File: Model/Task.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace ReminderApi.Model
{
    public class Task : BaseEntity
    {
        public Task()
        {

        }

        public Task(int toDoListId, string name, string note, DateTime? dueDate, DateTime? remindDate, bool isCompleted)
        {
            ToDoListId = toDoListId;
            Name = name;
            Note = note;
            DueDate = dueDate;
            RemindDate = remindDate;
            IsCompleted = isCompleted;
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public int ToDoListId { get; set; }
        public string Note { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime? RemindDate { get; set; }
        public bool IsCompleted { get; set; }
        public ToDoList ToDoList { get; set; }
    }
}

File: src/app/Models/task.ts
import { ToDoList } from './todolist';

export class Task {
  Id: number;
  Name: string;
  ToDoListId: number;
  Note: string;
  DueDate: Date;
  RemindDate: Date;
  IsCompleted: boolean;
  ToDoList: ToDoList;
  DateCreated: Date;
  DateModified: Date;

  public populate(res: any, todo: ToDoList) {
    this.Id = res.id;
    this.Name = res.name;
    this.ToDoListId = res.toDoListId;
    this.Note = res.note;
    this.DueDate = res.dueDate;
    this.RemindDate = res.remindDate;
    this.IsCompleted = res.isCompleted;
    this.DateCreated = res.dateCreated;
    this.DateModified = res.dateModified;
    return this;
  }
}
